include/synapticon_ros2_control/synapticon_interface.hpp

```diff
@@ -183,6 +183,9 @@ class SynapticonSystemInterface : public hardware_interface::SystemInterface {
   std::atomic<bool> in_normal_op_mode_ = false;
   // During spring adjust, don't allow control mode to change until the target position is reached
   std::atomic<bool> allow_mode_change_ = true;
+  // Spring adjust derivative term variables
+  double error_prev_;
```

Comment:
Need to initialize this or at least, don't use it if it's not initialized.

```suggestion
  std::optional<double> error_prev_;
```